commit 09bc748060aad36c41f5d6afee4408b5832b670a
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sat Aug 6 09:21:48 2022 +0900

    add country to index.ts

diff --git a/server/src/index.ts b/server/src/index.ts
index fd7197d..12efd40 100644
--- a/server/src/index.ts
+++ b/server/src/index.ts
@@ -1,6 +1,6 @@
 import { ApolloServer, gql } from "apollo-server";
 import * as fs from "fs";
-import { EmailAddressResolver } from "graphql-scalars";
+import { CountryCodeResolver, EmailAddressResolver } from "graphql-scalars";
 import { Query, Resolvers } from "./generated/graphql";
 
 const typeDefs = gql`
@@ -16,6 +16,9 @@ const resolvers: Resolvers<LoadingDataContext> = {
     me(_parent, _args, context, _info) {
       return context.Query.me;
     },
+    search(_parent, _args, context, _info) {
+      return context.Query.search;
+    },
   },
   Person: {
     name(parent, _args, _context, _info) {
@@ -24,8 +27,12 @@ const resolvers: Resolvers<LoadingDataContext> = {
     emailAddress(parent, _args, _context, _info) {
       return parent.emailAddress;
     },
+    country(parent, _args, _context, _info) {
+      return parent.country;
+    },
   },
   EmailAddress: EmailAddressResolver,
+  CountryCode: CountryCodeResolver,
 };
 
 const readJsonFile = async (relativeFileName: string): Promise<any> => {
