commit 12e6f2be420be9c3ab7399d21b838839bccba510
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sat Aug 6 09:05:52 2022 +0900

    Update codegen.yml to set EmailAddress as string

diff --git a/server/codegen.yml b/server/codegen.yml
index 0de22f5..a18a936 100644
--- a/server/codegen.yml
+++ b/server/codegen.yml
@@ -8,6 +8,8 @@ generates:
       - "typescript-resolvers"
     config:
       avoidOptionals: true
+      scalars:
+        EmailAddress: string
 hooks:
   afterOneFileWrite:
     - npx prettier --write
diff --git a/server/src/generated/graphql.ts b/server/src/generated/graphql.ts
index 5072f36..0270e8f 100644
--- a/server/src/generated/graphql.ts
+++ b/server/src/generated/graphql.ts
@@ -21,7 +21,7 @@ export type Scalars = {
   Boolean: boolean;
   Int: number;
   Float: number;
-  EmailAddress: any;
+  EmailAddress: string;
 };
 
 export type Person = {
