commit 1acf85ab43ce749efd0e72ebcbca5985d74651c7
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Wed Aug 3 00:17:09 2022 +0900

    use underscores in names to avoid warnings

diff --git a/server/src/index.ts b/server/src/index.ts
index 49f3b8b..3b884de 100644
--- a/server/src/index.ts
+++ b/server/src/index.ts
@@ -1,11 +1,6 @@
 import { ApolloServer, gql } from "apollo-server";
 import * as fs from "fs";
-import {
-  EmailAddressResolver,
-  TimestampResolver,
-  URLResolver,
-} from "graphql-scalars";
-import { EmailAddress } from "graphql-scalars/mocks";
+import { EmailAddressResolver } from "graphql-scalars";
 import { Query, Resolvers } from "./generated/graphql";
 
 const typeDefs = gql`
@@ -18,15 +13,15 @@ interface LoadingDataContext {
 
 const resolvers: Resolvers<LoadingDataContext> = {
   Query: {
-    me(parent, args, context, info) {
+    me(_parent, _args, _context, _info) {
       return null;
     },
   },
   Person: {
-    name(parent, args, context, info) {
+    name(parent, _args, _context, _info) {
       return parent.name;
     },
-    emailAddress(parent, args, context, info) {
+    emailAddress(parent, _args, _context, _info) {
       return parent.emailAddress;
     },
   },
