commit 1fb41bb9536e8c1e2f5a2de011ec9e45a01ccdc3
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sat Aug 6 22:04:38 2022 +0900

    update step 2

diff --git a/steps/2-graphql-scalars.sh b/steps/2-graphql-scalars.sh
index 8d65b5a..d713acb 100644
--- a/steps/2-graphql-scalars.sh
+++ b/steps/2-graphql-scalars.sh
@@ -3,12 +3,13 @@
 cd "$(dirname "$0")" || exit # REMOVE THIS IN aggregate.sh
 cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repository root
 
+# ## graphql-scalarsをfield型に使った際の動作確認
+
 # :large_orange_diamond: Action: 新しいターミナルを立ち上げてください。
 
 # ```:terminal
 # shellcheck disable=SC2164 # REMOVE THIS IN aggregate.sh
-cd server
-npm install graphql-scalars
+(cd server && npm install graphql-scalars)
 # ```
 
 # :large_orange_diamond: Action: 以下のコマンドを入力してください。
@@ -184,5 +185,79 @@ git apply patches/e60bb3a.patch # Update codegen.yml to set EmailAddress as stri
 
 # </div></details>
 
-run generate
-npm start
+# 上記の変更によって、型チェックが働きます。stringが期待されるemailAddressのResolverでnumberをreturnすると、エラーが表示されることが確認できます。
+
+# <details><summary>:white_check_mark: Result: エラーの確認</summary><div>
+
+# ```terminal
+# Type 'number' is not assignable to type 'Maybe<ResolverTypeWrapper<string>> | Promise<Maybe<ResolverTypeWrapper<string>>>'
+# The expected type comes from property 'emailAddress' which is declared here on type 'PersonResolvers<LoadingDataContext, Person>'
+# ```
+
+# ![2022-08-06_21h33_37.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/75738/11ef1c95-72e1-1ce8-446b-dd61a50cfb23.png)
+
+# ---
+
+# </div></details>
+
+# しかし、上記のようなgraphql-scalarsが提供する機能のみを使った型チェックは:
+
+#   - stringが期待されるところでnumberをreturnするようなエラーを検出できます
+#   - しかし、stringではあるものの、EmailAddressの形式として間違っているものはエラーにはなりません
+
+# それを確認しましょう。
+
+
+# :large_orange_diamond: Action: 以下のコマンドを入力してください。
+
+# ```shell
+git apply patches/4efc7a3.patch # wrong email address format passes type checking
+# ```
+
+# <details><summary>:white_check_mark: Result: 上記コマンドで更新される index.ts</summary><div>
+
+# ```diff:server/src/index.ts
+# emailAddress(parent, _args, _context, _info) {
+# -  return 10;
+# +  return "jason.summerwinter@@@@gmail.com";
+# }
+# ```    
+
+# ---
+
+# </div></details>
+
+# こちらはランタイムエラーでのみ検出可能になります。
+
+# <details><summary>:white_check_mark: Result: Apollo Studio Explorerでランタイムエラーを確認</summary><div>
+
+# https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/75738/d5bb9f73-4e20-3c32-9eeb-41e5e4d6c055.png
+
+# ---
+
+# </div></details>
+
+# :::note info
+# 静的型チェックによってEmailAddressに変換できない形式のstringをエラーとして検出するには、このチュートリアルのあとの方で紹介する独自typeの定義を使ったテクニックが必要です。
+# :::
+
+# それでは、graphql-scalarsが提供する機能のみを使った場合の型チェックの動作がわかったので、emailAddressをQuery.jsonファイルから値を取得する形に戻します。
+
+# :large_orange_diamond: Action: 以下のコマンドを入力してください。
+
+# ```shell
+git apply patches/f9b9f77.patch # revert the emailAddress back to parent.emailAddress
+# ```
+
+# <details><summary>:white_check_mark: Result: 上記コマンドで更新される index.ts</summary><div>
+
+# ```diff:server/src/index.ts
+# emailAddress(parent, _args, _context, _info) {
+# -  return "jason.summerwinnter@@@@gmail.com";
+# +  return 10;
+# }
+# ```    
+
+# ---
+
+# </div></details>
diff --git a/steps/3-argument-type.sh b/steps/3-argument-type.sh
index c8d11d7..bd5e398 100755
--- a/steps/3-argument-type.sh
+++ b/steps/3-argument-type.sh
@@ -3,6 +3,8 @@
 cd "$(dirname "$0")" || exit # REMOVE THIS IN aggregate.sh
 cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repository root
 
+# ## graphql-scalarsをargument型に使った際の動作確認
+
 # argument type
 #   Apollo Studio, send wrong type - error
 #   Apollo Studio, send wrong format - error
\ No newline at end of file
