commit 26ae80f82f6134aad424869d46165693a20d0343
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sat Aug 6 11:36:12 2022 +0900

    correct email format

diff --git a/server/src/index.ts b/server/src/index.ts
index 4d7df9e..0cf062f 100644
--- a/server/src/index.ts
+++ b/server/src/index.ts
@@ -10,7 +10,7 @@ const typeDefs = gql`
 
 const loadEmailDeepInsideServer = (): EmailAddressString => {
   // somewhere like database, deep inside the server side...
-  const valueFromDatabase = "jason.summerwinnter@@@@gmail.com";
+  const valueFromDatabase = "jason.summerwinnter@gmail.com";
   if (isEmailAddressString(valueFromDatabase)) {
     return valueFromDatabase;
   } else {
