commit 3288f624bf40302ff7ac51f4acc7f09bdcd98d6f
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sat Aug 6 10:06:04 2022 +0900

    Use EmailAddressString in codegen.yml

diff --git a/server/codegen.yml b/server/codegen.yml
index a18a936..012e564 100644
--- a/server/codegen.yml
+++ b/server/codegen.yml
@@ -6,10 +6,12 @@ generates:
     plugins:
       - "typescript"
       - "typescript-resolvers"
+      - "add":
+          content: "import * as myTypes from '../myTypes'"
     config:
       avoidOptionals: true
       scalars:
-        EmailAddress: string
+        EmailAddress: myTypes.EmailAddressString
 hooks:
   afterOneFileWrite:
     - npx prettier --write
diff --git a/server/src/generated/graphql.ts b/server/src/generated/graphql.ts
index 01cc175..af3eae6 100644
--- a/server/src/generated/graphql.ts
+++ b/server/src/generated/graphql.ts
@@ -3,6 +3,7 @@ import {
   GraphQLScalarType,
   GraphQLScalarTypeConfig,
 } from "graphql";
+import * as myTypes from "../myTypes";
 export type Maybe<T> = T | null;
 export type InputMaybe<T> = Maybe<T>;
 export type Exact<T extends { [key: string]: unknown }> = {
@@ -22,7 +23,7 @@ export type Scalars = {
   Int: number;
   Float: number;
   CountryCode: any;
-  EmailAddress: string;
+  EmailAddress: myTypes.EmailAddressString;
 };
 
 export type Person = {
