commit 3396121dde6f3ced560498c3c07e84b7d1d0d03e
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Wed Aug 3 00:23:15 2022 +0900

    wrong email format but passes type checking

diff --git a/server/src/index.ts b/server/src/index.ts
index 038b73a..f3d449c 100644
--- a/server/src/index.ts
+++ b/server/src/index.ts
@@ -22,7 +22,7 @@ const resolvers: Resolvers<LoadingDataContext> = {
       return parent.name;
     },
     emailAddress(parent, _args, _context, _info) {
-      return 10;
+      return "someone@@@@@gmail.com"; //wrong email format but passes type checking!
     },
   },
   EmailAddress: EmailAddressResolver,
diff --git a/steps/2-setup-codegen.sh b/steps/2-setup-codegen.sh
index 24e778b..f436a82 100644
--- a/steps/2-setup-codegen.sh
+++ b/steps/2-setup-codegen.sh
@@ -6,7 +6,7 @@ cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repos
 # npm install graphql-scalars
 # EmailAddress - schema.gql, data/Query.json, Apollo Studio
 #   override in index.ts, and return 10 -> error https://github.com/richardimaoka/tutorial-graphql-scalars/commit/5edea8ed55cf85d145edfc3c36746a74378aa148
-#   config.yml to set string -> static type checking!
+#   config.yml to set string -> static type checking! https://github.com/richardimaoka/tutorial-graphql-scalars/commit/5edea8ed55cf85d145edfc3c36746a74378aa148
         # ![2022-08-03_00h20_53.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/75738/25250495-3c88-b8e3-20f5-c89dca067c1e.png)
         # (property) emailAddress: Resolver<Maybe<ResolverTypeWrapper<string>>, Person, LoadingDataContext, {}>
         # Type '(parent: Person, _args: {}, _context: LoadingDataContext, _info: GraphQLResolveInfo) => number' is not assignable to type 'Resolver<Maybe<ResolverTypeWrapper<string>>, Person, LoadingDataContext, {}>'.
