commit 3c89889231799fcf673a6436362283d6a5b8f271
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sat Aug 6 17:15:19 2022 +0900

    make search argument required

diff --git a/server/schema.gql b/server/schema.gql
index 81edf42..c33aa49 100644
--- a/server/schema.gql
+++ b/server/schema.gql
@@ -9,5 +9,5 @@ type Person {
 
 type Query {
   me: Person
-  search(country: CountryCode): [Person]
+  search(country: CountryCode!): [Person]
 }
diff --git a/server/src/generated/graphql.ts b/server/src/generated/graphql.ts
index af3eae6..3b6616c 100644
--- a/server/src/generated/graphql.ts
+++ b/server/src/generated/graphql.ts
@@ -15,6 +15,9 @@ export type MakeOptional<T, K extends keyof T> = Omit<T, K> & {
 export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & {
   [SubKey in K]: Maybe<T[SubKey]>;
 };
+export type RequireFields<T, K extends keyof T> = Omit<T, K> & {
+  [P in K]-?: NonNullable<T[P]>;
+};
 /** All built-in and custom scalars, mapped to their actual values */
 export type Scalars = {
   ID: string;
@@ -40,7 +43,7 @@ export type Query = {
 };
 
 export type QuerySearchArgs = {
-  country: InputMaybe<Scalars["CountryCode"]>;
+  country: Scalars["CountryCode"];
 };
 
 export type ResolverTypeWrapper<T> = Promise<T> | T;
@@ -205,7 +208,7 @@ export type QueryResolvers<
     Maybe<Array<Maybe<ResolversTypes["Person"]>>>,
     ParentType,
     ContextType,
-    Partial<QuerySearchArgs>
+    RequireFields<QuerySearchArgs, "country">
   >;
 };
 
