commit 5edea8ed55cf85d145edfc3c36746a74378aa148
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Wed Aug 3 00:18:06 2022 +0900

    return 10 in EmailAddress

diff --git a/server/src/index.ts b/server/src/index.ts
index 3b884de..038b73a 100644
--- a/server/src/index.ts
+++ b/server/src/index.ts
@@ -22,7 +22,7 @@ const resolvers: Resolvers<LoadingDataContext> = {
       return parent.name;
     },
     emailAddress(parent, _args, _context, _info) {
-      return parent.emailAddress;
+      return 10;
     },
   },
   EmailAddress: EmailAddressResolver,
diff --git a/steps/2-setup-codegen.sh b/steps/2-setup-codegen.sh
index 77e2d35..b45dadb 100644
--- a/steps/2-setup-codegen.sh
+++ b/steps/2-setup-codegen.sh
@@ -5,7 +5,7 @@ cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repos
 
 # npm install graphql-scalars
 # EmailAddress - schema.gql, data/Query.json, Apollo Studio
-#   not from Query.json, but override in index.ts, and return 10 -> error
+#   override in index.ts, and return 10 -> error
 #   config.yml to set string -> static type checking!
 #   however, returning a string with wrong-format passes static typing...
 
