commit 6d06b652a00aa6a527d517579776707aac524e89
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sat Aug 6 17:20:22 2022 +0900

    revert back index.ts

diff --git a/server/src/index.ts b/server/src/index.ts
index 7747adf..12efd40 100644
--- a/server/src/index.ts
+++ b/server/src/index.ts
@@ -2,24 +2,11 @@ import { ApolloServer, gql } from "apollo-server";
 import * as fs from "fs";
 import { CountryCodeResolver, EmailAddressResolver } from "graphql-scalars";
 import { Query, Resolvers } from "./generated/graphql";
-import { EmailAddressString, isEmailAddressString } from "./myTypes";
 
 const typeDefs = gql`
   ${fs.readFileSync(__dirname.concat("/../schema.gql"), "utf8")}
 `;
 
-const loadEmailDeepInsideServer = (): EmailAddressString => {
-  // somewhere like database, deep inside the server side...
-  const valueFromDatabase = "jason.summerwinnter@gmail.com";
-  if (isEmailAddressString(valueFromDatabase)) {
-    return valueFromDatabase;
-  } else {
-    throw new TypeError(
-      `value from database = ${valueFromDatabase} is not a valid email address`
-    );
-  }
-};
-
 interface LoadingDataContext {
   Query: Query;
 }
@@ -37,17 +24,8 @@ const resolvers: Resolvers<LoadingDataContext> = {
     name(parent, _args, _context, _info) {
       return parent.name;
     },
-    emailAddress(_parent, _args, _context, _info) {
-      try {
-        const email = loadEmailDeepInsideServer();
-        return email;
-      } catch (error) {
-        //the server log only, not exposing the internal error detail to the API caller
-        console.log(error);
-        throw new Error(
-          "Internal Error occurred: could not retrieve emailAddress"
-        );
-      }
+    emailAddress(parent, _args, _context, _info) {
+      return parent.emailAddress;
     },
     country(parent, _args, _context, _info) {
       return parent.country;
