commit 81a0cb5a37b08e1f4b53df40f162f3f41e99c4e7
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Mon Aug 8 00:54:08 2022 +0900

    update step 5

diff --git a/steps/2-graphql-scalars.sh b/steps/2-graphql-scalars.sh
index 849904b..72ddc02 100644
--- a/steps/2-graphql-scalars.sh
+++ b/steps/2-graphql-scalars.sh
@@ -246,7 +246,7 @@ git apply patches/4efc7a3.patch # wrong email address format passes type checkin
 # </div></details>
 
 # :::note info
-# TypeScriptの静的型チェックによってEmailAddressに変換できない形式のstringをエラーとして検出するには、このチュートリアルのあとの方で紹介する独自typeの定義を使ったテクニックが必要です。
+# TypeScriptの静的型チェックによってEmailAddressに変換できない形式のstringをエラーとして検出するには、このチュートリアルの5. および 6. で紹介する独自型の定義を使ったテクニックが必要です。
 # :::
 
 # それでは、graphql-scalarsが提供する機能のみを使った場合の型チェックの動作がわかったので、emailAddressをQuery.jsonファイルから値を取得する形に戻します。
diff --git a/steps/4-argument-type.sh b/steps/4-argument-type.sh
index 2909e70..4f97751 100755
--- a/steps/4-argument-type.sh
+++ b/steps/4-argument-type.sh
@@ -11,6 +11,8 @@ cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repos
 
 # ```shell
 git apply patches/7833b58.patch # add CountryCode to schema.gql
+git apply patches/c1d6bcd.patch # add country to index.ts
+git apply patches/3f8aece.patch # update data/Query.json
 # ```
 
 # <details><summary>:white_check_mark: Result: 上記コマンドで更新される schema.gql</summary><div>
@@ -35,12 +37,6 @@ git apply patches/7833b58.patch # add CountryCode to schema.gql
 
 # </div></details>
 
-# :large_orange_diamond: Action: 以下のコマンドを入力してください。
-
-# ```shell
-git apply patches/c1d6bcd.patch # add country to index.ts
-# ```
-
 # <details><summary>:white_check_mark: Result: 上記コマンドで更新される index.ts</summary><div>
 
 # ```diff:server/src/index.ts
@@ -80,12 +76,6 @@ git apply patches/c1d6bcd.patch # add country to index.ts
 
 # </div></details>
 
-# :large_orange_diamond: Action: 以下のコマンドを入力してください。
-
-# ```shell
-git apply patches/3f8aece.patch # update data/Query.json
-# ```
-
 # <details><summary>:white_check_mark: Result: 上記コマンドで更新される Query.json</summary><div>
 
 # ```diff:server/data/Query.json
diff --git a/steps/5-optional.sh b/steps/5-optional.sh
index 56d60a1..c4e890f 100755
--- a/steps/5-optional.sh
+++ b/steps/5-optional.sh
@@ -4,26 +4,44 @@ cd "$(dirname "$0")" || exit # REMOVE THIS IN aggregate.sh
 cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repository root
 
 
-# ## 5. 
+# ## 5. (Optional) Custom scalarであるEmailAddressの情報を復元するため、TypeScriptで独自型を利用する
+
+# 本チュートリアル「2. graphql-scalarsをfield型に使った際の動作確認」の最後で、TypeScriptの型チェックを効かせて不正なメールアドレスの形式を検出するには、独自型の利用が必要なことを述べました。以下ではそのテクニックを実際に見ていきます。
 
 # :large_orange_diamond: Action: 以下のコマンドを入力してください。
 
 # ```shell
-npm install --save-dev @graphql-codegen/add 
+git apply patches/a827496.patch # EmailAddressString type
 # ```
 
-# :large_orange_diamond: Action: 以下のコマンドを入力してください。
+# <details><summary>:white_check_mark: Result: 上記コマンドで独自型 type EmailAddressString を定義するため、src/myTypes.tsを作成します。</summary><div>
 
-# ```shell
-git apply patches/a827496.patch # EmailAddressString type
+# ```typescript:src/myTypes.ts
+# export type EmailAddressString = string & { __type: "EmailAddressString" };
+
+# export const isEmailAddressString = (
+#   str: string
+# ): str is EmailAddressString => {
+#   const EMAIL_ADDRESS_REGEX =
+#     /^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+
+#   return EMAIL_ADDRESS_REGEX.test(str);
+# };
 # ```
 
+# ---
+
+# </div></details>
+
+
 # :large_orange_diamond: Action: 以下のコマンドを入力してください。
 
 # ```shell
+npm install --save-dev @graphql-codegen/add 
 git apply patches/3288f62.patch # Use EmailAddressString in codegen.yml
 # ```
 
+
 # :large_orange_diamond: Action: 以下のコマンドを入力してください。
 
 # ```shell
