commit 8c0a82a90eede4ad760373b9a321c2d05410165e
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sat Aug 6 17:49:38 2022 +0900

    use CountryString in generated code

diff --git a/server/codegen.yml b/server/codegen.yml
index 012e564..7c2e41b 100644
--- a/server/codegen.yml
+++ b/server/codegen.yml
@@ -12,6 +12,7 @@ generates:
       avoidOptionals: true
       scalars:
         EmailAddress: myTypes.EmailAddressString
+        CountryCode: myTypes.CountryString
 hooks:
   afterOneFileWrite:
     - npx prettier --write
diff --git a/server/src/generated/graphql.ts b/server/src/generated/graphql.ts
index 3b6616c..124c11d 100644
--- a/server/src/generated/graphql.ts
+++ b/server/src/generated/graphql.ts
@@ -25,7 +25,7 @@ export type Scalars = {
   Boolean: boolean;
   Int: number;
   Float: number;
-  CountryCode: any;
+  CountryCode: myTypes.CountryString;
   EmailAddress: myTypes.EmailAddressString;
 };
 
