commit 8d40ae6747d4f03cb0159cc05773694969b37a44
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Tue Aug 2 23:54:15 2022 +0900

    tutorial scripts

diff --git a/scripts/aggregate.sh b/scripts/aggregate.sh
new file mode 100755
index 0000000..904ff19
--- /dev/null
+++ b/scripts/aggregate.sh
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+cd "$(dirname "$0")" || exit 
+cd ../ || exit # cd to the git repository root
+
+AGGREGATED=$(find steps/* -print0 | \
+  xargs -0 cat | \
+  grep -v '#!/bin/sh' | \
+  grep -v '# REMOVE THIS' | \
+  sed 's/^\#\s//') 
+
+# copy README.md to the clipboard
+if which clip.exe 1>/dev/null # WSL
+then
+  echo "$AGGREGATED" | clip.exe 
+elif which pbcopy 1>/dev/null # Mac
+then
+  echo "$AGGREGATED" | pbcopy
+elif which xsel 1>/dev/null # Debian/Ubuntu Linux
+then
+  # WSL gives error on xsel as "xsel: Can't open display: (null)" so, using clip.exe on WSL
+  echo "$AGGREGATED" | xsel --clipboard --input
+fi
\ No newline at end of file
diff --git a/scripts/git-apply-print.sh b/scripts/git-apply-print.sh
new file mode 100755
index 0000000..fc55a9e
--- /dev/null
+++ b/scripts/git-apply-print.sh
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+cd "$(dirname "$0")" || exit 
+cd ../ || exit # cd to the git repository root
+
+git log --oneline --reverse | sed -r 's/(^\w+\b)/git apply patches\/\1.patch #/'
+
+# if you want to start from a certain commit (e.g.) dc61691
+# > git log --oneline --reverse dc61691 | sed -r 's/(^\w+\b)/git apply patches\/\1.patch #/'
diff --git a/scripts/git-patch-gen.sh b/scripts/git-patch-gen.sh
new file mode 100755
index 0000000..48f8c57
--- /dev/null
+++ b/scripts/git-patch-gen.sh
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+cd "$(dirname "$0")" || exit 
+cd ../ || exit # cd to the git repository root
+
+git log --oneline --reverse | \
+   sed -r 's/(^\w+\b)(.*)/git show --binary --patch \1 > patches\/\1.patch/'
+
+# if you want to start from a certain commit (e.g.) dc61691
+# > git log --oneline --reverse dc61691 | \
+# >   sed -r 's/(^\w+\b)(.*)/git show --binary --patch \1 > patches\/\1.patch/'
