commit 9a08842e4e71e735f4b57aa4e886378fbc75ae53
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Mon Aug 8 00:39:47 2022 +0900

    ok up to step 3

diff --git a/steps/2-graphql-scalars.sh b/steps/2-graphql-scalars.sh
index 1972d75..849904b 100644
--- a/steps/2-graphql-scalars.sh
+++ b/steps/2-graphql-scalars.sh
@@ -239,7 +239,7 @@ git apply patches/4efc7a3.patch # wrong email address format passes type checkin
 
 # <details><summary>:white_check_mark: Result: Apollo Studio Explorerでランタイムエラーを確認</summary><div>
 
-# https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/75738/d5bb9f73-4e20-3c32-9eeb-41e5e4d6c055.png
+# ![2022-08-08_00h28_44.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/75738/d0bbfce6-f18a-582c-5896-ec953f159c8d.png)
 
 # ---
 
diff --git a/steps/3-many-custom-scalars.sh b/steps/3-many-custom-scalars.sh
index d86d70d..bc66890 100644
--- a/steps/3-many-custom-scalars.sh
+++ b/steps/3-many-custom-scalars.sh
@@ -38,7 +38,6 @@ git apply patches/20f9644.patch # many custom scalars
 
 # </div></details>
 
-
 # <details><summary>:white_check_mark: Result: 上記コマンドで更新される schema.gql</summary><div>
 
 # ```diff:server/schema.gql
@@ -147,6 +146,7 @@ git apply patches/20f9644.patch # many custom scalars
 
 # <details><summary>:white_check_mark: Result: Apollo Studio Explorerで確認</summary><div>
 
+# ![2022-08-08_00h37_18.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/75738/93190d7f-e8e2-ebd4-6d0f-9c1497bec43b.png)
 
 # ---
 
