commit 9be031c40d83bb1805bb43c7d07f61a023833c78
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Wed Aug 3 00:26:12 2022 +0900

    update Query.json

diff --git a/server/data/Query.json b/server/data/Query.json
index de7be41..a441dce 100644
--- a/server/data/Query.json
+++ b/server/data/Query.json
@@ -1,4 +1,3 @@
 {
-  "hello": "Hello World",
-  "url": "https:||google.com"
+  "me": { "name": "Jason Summerwinter" }
 }
diff --git a/server/src/index.ts b/server/src/index.ts
index f3d449c..acc5a22 100644
--- a/server/src/index.ts
+++ b/server/src/index.ts
@@ -13,8 +13,8 @@ interface LoadingDataContext {
 
 const resolvers: Resolvers<LoadingDataContext> = {
   Query: {
-    me(_parent, _args, _context, _info) {
-      return null;
+    me(_parent, _args, context, _info) {
+      return context.Query.me;
     },
   },
   Person: {
diff --git a/steps/2-setup-codegen.sh b/steps/2-setup-codegen.sh
index f436a82..921e6dd 100644
--- a/steps/2-setup-codegen.sh
+++ b/steps/2-setup-codegen.sh
@@ -13,7 +13,7 @@ cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repos
         #   Type '(parent: Person, _args: {}, _context: LoadingDataContext, _info: GraphQLResolveInfo) => number' is not assignable to type 'ResolverFn<Maybe<ResolverTypeWrapper<string>>, Person, LoadingDataContext, {}>'.
         #     Type 'number' is not assignable to type 'Maybe<ResolverTypeWrapper<string>> | Promise<Maybe<ResolverTypeWrapper<string>>>'.ts(2322)
         # graphql.ts(170, 3): The expected type comes from property 'emailAddress' which is declared here on type 'PersonResolvers<LoadingDataContext, Person>'
-#   however, returning a string with wrong-format passes static typing...
+#   however, returning a string with wrong-format passes static typing... https://github.com/richardimaoka/tutorial-graphql-scalars/commit/3396121dde6f3ced560498c3c07e84b7d1d0d03e
 
 # argument type
 #   Apollo Studio, send wrong type - error
