commit eb0f7f2b4eb1b3cab03d45299cc7d011d92790be
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sun Jul 31 16:11:58 2022 +0900

    updated

diff --git a/steps/2-setup-codegen.sh b/steps/2-setup-codegen.sh
index 1afe9ea..77e2d35 100644
--- a/steps/2-setup-codegen.sh
+++ b/steps/2-setup-codegen.sh
@@ -13,7 +13,5 @@ cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repos
 #   Apollo Studio, send wrong type - error
 #   Apollo Studio, send wrong format - error
 
-
-
-npm run generate
+run generate
 npm start
diff --git a/steps/3-optional.sh b/steps/3-optional.sh
new file mode 100755
index 0000000..5bb2dcd
--- /dev/null
+++ b/steps/3-optional.sh
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+cd "$(dirname "$0")" || exit # REMOVE THIS IN aggregate.sh
+cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repository root
+
+git apply # restore the EmailAddress information from the string type in JavaScript
+
+# if you use a text schema definition, you can't use ScalrNameTypeDefinition
+
+# nested structure
+#   https://www.graphql-scalars.dev/docs/quick-start
+#   type Person {
+#     birthDate: DateTime
+#     ageInYears: PositiveInt
+#     heightInInches: PositiveFloat
+#     minimumHourlyRate: NonNegativeFloat
+#     currentlyActiveProjects: NonNegativeInt
+#     email: EmailAddress
+#     homePage: URL
+#     phoneNumber: PhoneNumber
+#     homePostalCode: PostalCode
+#    }
+#
+#    or types for realistic apps, like Reviewer and Store, etc
+
+# many more scalar types
+
+# if you use .gql schema file, you can't use ScalrNameTypeDefinition
+
+npm
\ No newline at end of file
diff --git a/steps/3-scalar.sh b/steps/3-scalar.sh
deleted file mode 100755
index d43b197..0000000
--- a/steps/3-scalar.sh
+++ /dev/null
@@ -1,14 +0,0 @@
-#!/bin/sh
-
-cd "$(dirname "$0")" || exit # REMOVE THIS IN aggregate.sh
-cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repository root
-
-npm install graphql-scalars
-
-git apply # EmailAddress in schema, resolver
-# query in apollo studio, error due to wrong return format
-
-git apply # argument type in schema, resolver
-# query in apollo studio, errors
-
-# if you use .gql schema file, you can't use ScalrNameTypeDefinition
diff --git a/steps/4-optional.sh b/steps/4-optional.sh
deleted file mode 100755
index 9e9f616..0000000
--- a/steps/4-optional.sh
+++ /dev/null
@@ -1,13 +0,0 @@
-#!/bin/sh
-
-cd "$(dirname "$0")" || exit # REMOVE THIS IN aggregate.sh
-cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repository root
-
-git apply # restore the EmailAddress information from the string type in JavaScript
-
-# if you use a text schema definition, you can't use ScalrNameTypeDefinition
-
-# Store and Reviewer structure
-
-# many more scalar types
-
