commit f18619ccf695621d6fe9c05d90dfb1501f0e37a3
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Wed Aug 3 00:22:11 2022 +0900

    type checking

diff --git a/server/codegen.yml b/server/codegen.yml
index a777c66..a18a936 100644
--- a/server/codegen.yml
+++ b/server/codegen.yml
@@ -9,7 +9,7 @@ generates:
     config:
       avoidOptionals: true
       scalars:
-        Timestamp: string
+        EmailAddress: string
 hooks:
   afterOneFileWrite:
     - npx prettier --write
diff --git a/server/src/generated/graphql.ts b/server/src/generated/graphql.ts
index 5072f36..0270e8f 100644
--- a/server/src/generated/graphql.ts
+++ b/server/src/generated/graphql.ts
@@ -21,7 +21,7 @@ export type Scalars = {
   Boolean: boolean;
   Int: number;
   Float: number;
-  EmailAddress: any;
+  EmailAddress: string;
 };
 
 export type Person = {
diff --git a/steps/2-setup-codegen.sh b/steps/2-setup-codegen.sh
index b45dadb..24e778b 100644
--- a/steps/2-setup-codegen.sh
+++ b/steps/2-setup-codegen.sh
@@ -5,8 +5,14 @@ cd ../ || exit               # REMOVE THIS IN aggregate.sh - cd to the git repos
 
 # npm install graphql-scalars
 # EmailAddress - schema.gql, data/Query.json, Apollo Studio
-#   override in index.ts, and return 10 -> error
+#   override in index.ts, and return 10 -> error https://github.com/richardimaoka/tutorial-graphql-scalars/commit/5edea8ed55cf85d145edfc3c36746a74378aa148
 #   config.yml to set string -> static type checking!
+        # ![2022-08-03_00h20_53.png](https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/75738/25250495-3c88-b8e3-20f5-c89dca067c1e.png)
+        # (property) emailAddress: Resolver<Maybe<ResolverTypeWrapper<string>>, Person, LoadingDataContext, {}>
+        # Type '(parent: Person, _args: {}, _context: LoadingDataContext, _info: GraphQLResolveInfo) => number' is not assignable to type 'Resolver<Maybe<ResolverTypeWrapper<string>>, Person, LoadingDataContext, {}>'.
+        #   Type '(parent: Person, _args: {}, _context: LoadingDataContext, _info: GraphQLResolveInfo) => number' is not assignable to type 'ResolverFn<Maybe<ResolverTypeWrapper<string>>, Person, LoadingDataContext, {}>'.
+        #     Type 'number' is not assignable to type 'Maybe<ResolverTypeWrapper<string>> | Promise<Maybe<ResolverTypeWrapper<string>>>'.ts(2322)
+        # graphql.ts(170, 3): The expected type comes from property 'emailAddress' which is declared here on type 'PersonResolvers<LoadingDataContext, Person>'
 #   however, returning a string with wrong-format passes static typing...
 
 # argument type
