commit f9b9f77054e8f5e76658b104a1e55cb79fb4562e
Author: Richard Imaoka <richard.e.imaoka@gmail.com>
Date:   Sat Aug 6 09:08:31 2022 +0900

    revert the emailAddress back to parent.emailAddress

diff --git a/server/src/index.ts b/server/src/index.ts
index 436f279..8b2c280 100644
--- a/server/src/index.ts
+++ b/server/src/index.ts
@@ -22,7 +22,7 @@ const resolvers: Resolvers<LoadingDataContext> = {
       return parent.name;
     },
     emailAddress(parent, _args, _context, _info) {
-      return "jason.summerwinnter@@@@gmail.com";
+      return parent.emailAddress;
     },
   },
   EmailAddress: EmailAddressResolver,
